Code/Legacy/CrySystem/LevelSystem/SpawnableLevelSystem.cpp

```diff
@@ -589,6 +589,9 @@ namespace LegacyLevelSystem
 
         // Delete level entities and remove them from the game entity context
         AzFramework::RootSpawnableInterface::Get()->ReleaseRootSpawnable();
+        
+        // Process the queued deactivate calls for the unloaded entities
+        AzFramework::RootSpawnableInterface::Get()->ProcessSpawnableQueue();
```

Comment:
Should this instead use `ProcessSpawnableQueueUntilEmpty` to make sure everything is processed?
```suggestion
        AzFramework::RootSpawnableInterface::Get()->ProcessSpawnableQueueUntilEmpty();
```

Reply:
Based the description, the ProcessSpawnableQueue() should be good for the loading a new level use case
```

        //! Force processing all SpawnableEntitiesManager requests immediately
        //! This is useful when loading a different level while SpawnableEntitiesManager still has
        //! pending requests
        virtual void ProcessSpawnableQueue() = 0;
        //! Force processing all SpawnableEntitiesManager requests immediately and keep reprocessing until the queue is empty
        //! This is useful when unloading and shutting down to ensure that all pending requests are cleared.
        virtual void ProcessSpawnableQueueUntilEmpty() = 0;
```
Currently the ProcessSpawnableQueueUntilEmpty is only used at the system shutting down at this moment.

By looking at the code, it seems it's possible that some requests get re-queued. It might be safer to use ProcessSpawnableQueueUntilEmpty.